packages/quantic/force-app/main/default/lwc/quanticRecentResultsList/__tests__/quanticRecentResultList.test.js

```diff
@@ -0,0 +1,474 @@
+
+    it('should build the controller and subscribe to its state', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionMocks.buildRecentResultsList).toHaveBeenCalledTimes(1);
+      expect(functionMocks.buildRecentResultsList).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          initialState: {results: defaultResults},
+          options: expect.objectContaining({
+            maxLength: defaultMaxLength,
+          }),
+        }
+      );
+      expect(functionMocks.recentResultsListSubscriber).toHaveBeenCalledTimes(
+        1
+      );
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+      jest
+        .spyOn(mockHeadlessLoader, 'getHeadlessEnginePromise')
+        .mockImplementation(() => Promise.resolve(exampleEngine));
+      console.error = jest.fn();
+    });
+
+    afterEach(() => {
+      recentResultsListState = initialRecentResultsListState;
+      jest.clearAllMocks();
+    });
+
+    describe('when there are recent results', () => {
+      const exampleResults = mockResults;
+      beforeEach(() => {
+        recentResultsListState = {
+          ...recentResultsListState,
+          results: exampleResults,
+        };
+      });
+
+      it('should no longer display the placeholder component', async () => {
+        const element = createTestComponent();
+        updateRecentResultsState();
+        await flushPromises();
+
+        const placeholder = element.shadowRoot.querySelector(
+          selectors.placeholder
+        );
+
+        expect(placeholder).toBeNull();
+      });
+
+      it('should display the recent results list with the default options', async () => {
+        recentResultsListState = {
+          results: exampleResults,
+          maxLength: defaultMaxLength,
+        };
+        const element = createTestComponent();
+        updateRecentResultsState();
+        await flushPromises();
+
+        const recentResultsContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentResultsContainer).not.toBeNull();
+        expect(recentResultsContainer.title).toEqual(defaultRecentResultsTitle);
+
+        const recentResultsListItems = element.shadowRoot.querySelectorAll(
+          selectors.recentResultItem
+        );
+        expect(recentResultsListItems.length).toEqual(exampleResults.length);
```

Comment:
comparing just the length is not enough. we need to check that each individual recent result is displayed correctly with its correct options.